crates/flux-refineck/src/checker.rs

```diff
@@ -883,7 +883,9 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
                 && let Some(callee_def_id) = callee_def_id
                 && genv.def_kind(callee_def_id).is_fn_like()
             {
-                let callee_no_panic = genv.no_panic(callee_def_id);
+                let callee_no_panic = fn_sig.skip_binder_ref().skip_binder_ref().no_panic()
```

Comment:
To avoid skipping binders, we should move all this logic (everything inside `M::NAME == "refine"`) to after we instantiate the signature. I'd put it after `let mut at = infcx.at(span)`.